### DISCUSSION TITLE
Support for DS920+

### DISCUSSION THREAD

**Discussion by rinzes** [runfalk/synology-wireguard#75] [Ideas]
I tried looking on docker hub for your image but unable to find for my DS920+. I see you support the ds918+ so maybe you could add the DS920+ also

**Comment by runfalk** [comment]
This package is not published on docker hub (at least not by me). DS920+ is probably already supported, you should try with the Geminilake package (https://github.com/runfalk/synology-wireguard/releases).

**Reply by rinzes** [reply]
Do you know if this will supply wireguard support for a docker image. Also I note the DS9218+ has this:

DS918+ | apollolake | 6.2 | Yes
-- | -- | -- | --

**Reply by runfalk** [reply]
I don't know what you're asking for with the docker image. WireGuard is a kernel plugin, I don't think they can even run in Docker.

The table in the README gets updated if someone with that particular model has confirmed that their model works. Realistically all geminilake should be the same. DS918+ has an apollolake processor and DS920+ has a geminilake processor. This means that they can't run the same package.

Since you seem to be a bit new to this, I urge you to be cautious when using this package. You could lock yourself out from your NAS in the worst case if you don't know what you're doing.

**Reply by Brettdah** [reply]
Hello there I also have a 920+
and I think what @rinzes is saying is that there [is docker image that are running wiregard ](https://docs.linuxserver.io/images/docker-wireguard) at least today....
And I just tried to turn this up and I it fail [like it shourd](https://github.com/linuxserver/docker-wireguard/issues/122) the maintainer officially support only debian based distro....
: 
docker log show this
``` log
[...]
[cont-init.d] 10-adduser: exited 0.
[cont-init.d] 30-module: executing...
Uname info: Linux 4985f5c80923 4.4.180+ #42218 SMP Mon Oct 18 19:16:55 CST 2021 x86_64 x86_64 x86_64 GNU/Linux
RTNETLINK answers: Operation not supported
**** The wireguard module is not active, will attempt kernel header install and module compilation. ****
**** Attempting kernel header install ****
Get:1 http://archive.ubuntu.com/ubuntu bionic InRelease [242 kB]
Get:2 http://archive.ubuntu.com/ubuntu bionic-updates InRelease [88.7 kB]
Get:3 http://archive.ubuntu.com/ubuntu bionic-security InRelease [88.7 kB]
Get:4 http://archive.ubuntu.com/ubuntu bionic/universe Sources [11.5 MB]
Get:5 http://archive.ubuntu.com/ubuntu bionic/restricted Sources [5,823 B]
Get:6 http://archive.ubuntu.com/ubuntu bionic/multiverse Sources [216 kB]
Get:7 http://archive.ubuntu.com/ubuntu bionic/main Sources [1,063 kB]
Get:8 http://archive.ubuntu.com/ubuntu bionic/restricted amd64 Packages [13.5 kB]
Get:9 http://archive.ubuntu.com/ubuntu bionic/main amd64 Packages [1,344 kB]
Get:10 http://archive.ubuntu.com/ubuntu bionic/universe amd64 Packages [11.3 MB]
Get:11 http://archive.ubuntu.com/ubuntu bionic/multiverse amd64 Packages [186 kB]
Get:12 http://archive.ubuntu.com/ubuntu bionic-updates/multiverse Sources [18.2 kB]
Get:13 http://archive.ubuntu.com/ubuntu bionic-updates/main Sources [655 kB]
Get:14 http://archive.ubuntu.com/ubuntu bionic-updates/restricted Sources [30.2 kB]
Get:15 http://archive.ubuntu.com/ubuntu bionic-updates/universe Sources [598 kB]
Get:16 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 Packages [3,036 kB]
Get:17 http://archive.ubuntu.com/ubuntu bionic-updates/restricted amd64 Packages [840 kB]
Get:18 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 Packages [2,252 kB]
Get:19 http://archive.ubuntu.com/ubuntu bionic-updates/multiverse amd64 Packages [29.0 kB]
Get:20 http://archive.ubuntu.com/ubuntu bionic-security/main Sources [331 kB]
Get:21 http://archive.ubuntu.com/ubuntu bionic-security/multiverse Sources [8,784 B]
Get:22 http://archive.ubuntu.com/ubuntu bionic-security/restricted Sources [26.8 kB]
Get:23 http://archive.ubuntu.com/ubuntu bionic-security/universe Sources [365 kB]
Get:24 http://archive.ubuntu.com/ubuntu bionic-security/restricted amd64 Packages [806 kB]
Get:25 http://archive.ubuntu.com/ubuntu bionic-security/main amd64 Packages [2,596 kB]
Get:26 http://archive.ubuntu.com/ubuntu bionic-security/multiverse amd64 Packages [21.1 kB]
Get:27 http://archive.ubuntu.com/ubuntu bionic-security/universe amd64 Packages [1,474 kB]
Fetched 39.2 MB in 6s (6,387 kB/s)
Reading package lists...
**** No kernel headers found in the Ubuntu or Debian repos!! Will try the headers from host (if mapped), may or may not work ****
**** Kernel headers don't seem to be available, can't compile the module. Sleeping now. . . ****
```

I now you are not the maintainer of this image... but may be we can think of a way to produce a docker image for synology NAS that way we can avoid beeing locked out of our precious things ;)
May be using the Xpenology... but just right now I'm not really sure how to  start... cause for a container we will want a few package, and I think no GUI

Maybe an other way would be to provide the headers... but don't know what it would do to an ubuntu based image xD

**Reply by Brettdah** [reply]
but first I'll try with alpine xD

**Comment by ehcorn** [comment]
This package works on my DS920+, I had to install wireguard on the synology so it shows up in my package center before I can run docker containers that use wireguard. As far as I know this is because wireguard has to be available on the host OS kernel before it can be used in a docker container. I followed the install instructions and it works, maybe check to make sure you setup the startup code in the synology settings too.

**Reply by Dark1886** [reply]
You got this on the DS920+, did you have to do a full manual install or did a prebuilt version work?

**Reply by ehcorn** [reply]
I used the gemini lake build, installed through the software centre via manual install but I used the release build. At least that's what I thought I did, because when I checked the software center it shows that I've used a packge authored by Andreas Runfalk but the version is 1.0.2021606 which is different from what's in the published folder. Some searching lead me here: https://github.com/runfalk/synology-wireguard/issues/88#issuecomment-893363490 where it looks like someone's compiled a binary that might be the one I used. Comments lead me to believe this should have improved compatibility with the 920+ but I've been using wireguard successfully on my 920+ for several months now. hope that helps!
